ext/standard/var_unserializer.re

```diff
@@ -947,7 +947,13 @@ use_double:
 	} else if (len == 1) {
 		ZVAL_INTERNED_STR(rval, ZSTR_CHAR((zend_uchar)*str));
 	} else if (as_key) {
-		ZVAL_STR(rval, zend_string_init_interned(str, len, 0));
+		zend_string *key = zend_string_init(str, len, 0);
+		zend_string *ikey = zend_interned_string_find_permanent(key);
```

Comment:
This will only work for strings that are created during startup, which is probably not terribly useful. We'd be mainly interested in strings that are either interned per-request or part of opcache. I think this would require adding a new hook for this purpose that can be replaced by opcache.

Reply:
I don't think this ought to use interned strings at all.

It looks simpler to add a table to unserialize data for keys.

Reply:
I think the idea here is specifically to reuse normal interned strings for things like object property names, which are usually available as interned strings.

Maybe @dstogov can comment.

Reply:
With opcache ``zend_string_init_interned()`` doesn't create new interned strings (just search for existing ones).
To implement the same behavior  without opcache, we need to introduce a new callback in zend_string API.

Reply:
I don't think that we want to introduce a new callback in the stable PHP versions. So, postpone the fix to PHP 8.1?